crates/bevy_core_pipeline/src/postprocess/mod.rs

```diff
@@ -0,0 +1,484 @@
+        RenderPipelineDescriptor, Sampler, SamplerBindingType, SamplerDescriptor, Shader,
+        ShaderStages, ShaderType, SpecializedRenderPipeline, SpecializedRenderPipelines,
+        TextureDimension, TextureFormat, TextureSampleType,
+    },
+    renderer::{RenderContext, RenderDevice, RenderQueue},
+    texture::{BevyDefault, GpuImage, Image},
+    view::{ExtractedView, ViewTarget},
+    Render, RenderApp, RenderSet,
+};
+use bevy_utils::prelude::default;
+
+use crate::{
+    core_2d::graph::{Core2d, Node2d},
+    core_3d::graph::{Core3d, Node3d},
+    fullscreen_vertex_shader,
+};
+
+/// The handle to the built-in postprocessing shader `postprocess.wgsl`.
+const POSTPROCESSING_SHADER_HANDLE: Handle<Shader> = Handle::weak_from_u128(14675654334038973533);
+
+/// The handle to the default chromatic aberration lookup texture.
+///
+/// This is just a 3x1 image consisting of one red pixel, one green pixel, and
+/// one blue pixel, in that order.
+const DEFAULT_CHROMATIC_ABERRATION_LUT_HANDLE: Handle<Image> =
+    Handle::weak_from_u128(2199972955136579180);
+
+/// The default chromatic aberration intensity amount, in a fraction of the
+/// window size.
+const DEFAULT_CHROMATIC_ABBERATION_INTENSITY: f32 = 0.02;
+
+/// The default maximum number of samples for chromatic aberration.
+const DEFAULT_CHROMATIC_ABERRATION_MAX_SAMPLES: u32 = 8;
+
+/// The raw RGBA data for the default chromatic aberration gradient.
+///
+/// This consists of one red pixel, one green pixel, and one blue pixel, in that
+/// order.
+static DEFAULT_CHROMATIC_ABBERATION_LUT_DATA: [u8; 12] =
+    [255, 0, 0, 255, 0, 255, 0, 255, 0, 0, 255, 255];
+
+/// A plugin that implements a built-in postprocessing stack with some common
+/// effects.
+///
+/// Currently, this only consists of chromatic aberration.
+pub struct PostprocessingPlugin;
+
+/// Adds colored fringes to the edges of objects in the scene.
+///
+/// [Chromatic aberration] simulates the effect when lenses fail to focus all
+/// colors of light toward a single point. It causes rainbow-colored streaks to
+/// appear, which are especially apparent on the edges of objects. Chromatic
+/// aberration is commonly used for collision effects, especially in horror
+/// games.
+///
+/// Bevy's implementation is based on that of *Inside* ([Gjøl & Svendsen 2016]).
+/// It's based on a customizable lookup texture, which allows for changing the
+/// color pattern. By default, the color pattern is simply a 3×1 pixel texture
+/// consisting of red, green, and blue, in that order, but you can change it to
+/// any 1D image in order to achieve different effects.
+///
+/// [Chromatic aberration]: https://en.wikipedia.org/wiki/Chromatic_aberration
+///
+/// [Gjøl & Svendsen 2016]: https://github.com/playdeadgames/publications/blob/master/INSIDE/rendering_inside_gdc2016.pdf
+#[derive(Reflect, Component, Clone)]
+#[reflect(Component, Default)]
+pub struct ChromaticAberration {
```

Comment:
What's your plan for scaling this up. As in, how would we start adding more effects. Is the intention to use different component for each effect then toggle them using ifdef in the `post_processing` shader?

This doesn't need to be solved in this PR btw, I'm just wondering how you plan to add more effects.

Reply:
Yes, exactly. Each effect gets a different component, but they collapse into the same ubershader.